src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,177 @@
+package edu.wpi.grip.core;
+
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Point;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+
+/**
+ * Commonly used {@link SocketHint SocketHints}
+ */
+public final class SocketHints {
+
+    /**
+     * Provides some default domains
+     */
+    public enum Domain {
```

Comment:
@PaulaRudy What do you think? I wrote it because I thought that it would make it cleaner and easier to create domains.